### DISCUSSION TITLE
OTA From Personal Server

### DISCUSSION THREAD

**Discussion by smhmaqsood** [arendst/Tasmota#11187] [Q&A] [answered]
I've trouble regarding OTA through a private server. I'm using "http" (NOT "https") but still I'm unable to perform OTA from a private server. Can anyone tell me what parameters and settings should be considered while configuring Tasmota and to make it able to perform OTA through a private server?

**Comment by Jason2866** [comment, marked as answer]
This small one is working https://github.com/arendst/Tasmota/tree/development/tools/ota_server

**Reply by smhmaqsood** [reply]
Thanks For Reply.
Can you tell me what parameters or variables should I change to perform OTA from my Private cloud. I'm using URL based on "http".

**Reply by nagyrobi** [reply]
I have a basic Apache installed with no SSL and perform my own updates in-house. It's very handy to have the platformio-core and an apache server on the same machine, you can OTA directly from the build output directory if you allow apache to see it.

`#define OTA_URL` doesn't work from `user_config_override.h` but you can trigger OTA from the web interface by pasting in the correct URL of the bin.gz file.

**Reply by barbudor** [reply]
You can use basically any HTTP server.
I used to use `python -m http.server` but recently moved to `lighttpd`
Use only `*.bin.gz` files and don't forget to have `tasmota-minimal.bin.gz` if you happen to upload larger files.

`#define OTA_URL` works perfectly well from `user_config_override.h` if you stick by the rule and base your custom firmware on `tasmota(32)`.
I have `#define OTA_URL                "http://192.168.168.244:8000/latest/tasmota-" CODE_IMAGE_STR` in my `user_config_override.h` where `CODE_IMAGE_STR` defines my custom firmware name.

**Reply by sfromis** [reply]
I'm using a very rudimentary http server of my own creation, responding with headers like:
```
HTTP/1.0 200 Found
Date: Tue, 02 Mar 2021 16:11:00 +0100
Server: xxxxxx
Content-Length: 123456
Content-Type: application/x-gzip
```
Having length is important, at least.

**Reply by smhmaqsood** [reply]
Thank you all for your guidance.
